integration/components/image/integration_test.go

```diff
@@ -0,0 +1,39 @@
+// Copyright 2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package image
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"testing"
+)
+
+var (
+	integration      = flag.Bool("integration", false, "run integration tests")
```

Comment:
What's the purpose of this flag? Can't we test on `TestImage` is it's running as root and skip if not?

Reply:
This flag is need to avoid running these tests as part on unit tests: https://github.com/inspektor-gadget/inspektor-gadget/blob/main/Makefile#L237.

Reply:
> why we're differentiating these tests from "unit tests"

Maybe I am being too careful. Since we are spinning a registry container and rely on `docker` I think it makes sense to have it as integration tests. I believe "unit tests" should just work irrespective to what is installed on the node or not?

Reply:
Yeah, makes sense. Also this one modifies the host where it's run as it creates some gadget images. I still don't like this parameter, but I don't know how to improve the situation.

Reply:
> I still don't like this parameter, but I don't know how to improve the situation.

Just to understand, is this about how we implement the parameter? The other way I can think of is using build tags e.g `//go:build components` and running them as `go test -exec sudo -tags=components -v ./integration/components/...`

Reply:
If I do `cd integration/inspektor-gadget/ && go test -exec sudo .`, tests will be skipped, same with happen with build tags. I don't really know if this is a common approach.